connman/src/detect.c

```diff
@@ -27,7 +27,13 @@
 
 #include "connman.h"
 
+struct device_details {
+	enum connman_device_type type;
+	int index;
+};
+
 static GSList *device_list = NULL;
+static gint delayed_newlink_ids[MAX_CONNMAN_SERVICE_TYPES] = { 0 };
```

Comment:
Can we relay on there only being one new link per service type?

Reply:
Yes but this is more of an idea level approach still that might this be the cause even. Which is why I just added commit to add delayed only when there is no MAC. Didn't want to overdo this yet.

Reply:
Yep, this made things worse when there is no MAC set for a device. Removed.